sympy/utilities/lambdify.py

```diff
@@ -797,10 +806,19 @@ def _lambdifygenerated(x):
                            'allow_unknown_functions': True,
                            'user_functions': user_functions})
 
+
+    if  type(args) is set:
+        warnings.warn('WARNING: The list of arguments is a `set`. The outcome of enumerate(args) is used to `lambdify`, but its outcome is not predictable. Lambdify will stop here and return `False`', category=DeprecationWarning)
+        return False  # Anything else that block the code would be fine
```

Comment:
This should use SymPyDeprecationWarning like this:
https://github.com/sympy/sympy/blob/87cca59f3507e1052631da0819edd0cfa1baadc0/sympy/diffgeom/diffgeom.py#L254-L259

For now this can be deprecated and then it can be mad an error later. The issue argument should refer to an issue created on github that explains what is deprecated and why (a new issue should be created for that purpose).